### ISSUE TITLE
Nested transactions behavior

### ISSUE THREAD

**Issue by seriousbusinessuser** [igrishaev/pg2#12]
Nested transactions must be ignored, similar to how it's done in clojure jdbc:
https://github.com/clojure/java.jdbc/blob/master/src/main/clojure/clojure/java/jdbc.clj#L778
Currently top-level rollback doesn't work if some code deep down in the call stack commits its own nested transaction. This is really important.

**Comment by igrishaev** [comment]
Thank you for reporting that. I need to think on the issue and make some experiments.

Meanwhile, could you please give me some more context? For example, why would you have nested transactions? Why don't you have just single transaction that covers the whole logic?

**Comment by seriousbusinessuser** [comment]
@igrishaev Sure, it's easier to reuse code if you don't have to worry about some hidden transaction somewhere deep down in the code written by someone else. 

For example, one might want to add function which will save a blog post comment with a twist - commentary author's basic information must be saved as well. Since this hypothetical blogging app existed for some time already and basic information saving part has already been implemented in `save-basic-info` function, one is only required to write a function to save a blog post comment. The only catch is that `save-basic-info` internally uses a transaction. 

With JDBC we could just write something like this:
```clojure
(defn save-post-comment [data]
  (with-transaction [tx ds]
    (foo data)
    (save-basic-info (:info data))
    (bar data)))
```

With pg2 would need to pass `tx` as an optional argument to `save-basic-info` and check if we need to start a new transaction or use existing one. This adds more work where none is required with jdbc.

**Comment by igrishaev** [comment]
I'm reading this code sample and not getting why save-basic-info and other functions don't accept the tx parameter? How do they reach the connection to save the data? Do they rely on any kind of a global variable?

**Comment by seriousbusinessuser** [comment]
@igrishaev `save-basic-info` doesn't accept `tx` because it's used separately from `save-post-comment` and starts its own transaction:

> With pg2 we would need to pass tx as an optional argument to `save-basic-info` and check if we need to start a new transaction or use existing one. This adds more work where none is required with jdbc.

Other functions don't accept the `tx` parameter because it's an example and those function are not important, they're there just to illustrate some additional work being done inside `save-post-comment`.

**Comment by igrishaev** [comment]
Sorry for the pause, I've been traveling with my family. I'm still thinking on that, and most likely it can be fixed by a check if the current connect is already in transaction. I can extend the `with-connection` macro such that it check if there is already a transaction, and if yes, just skip the begin/commit clauses.

**Comment by igrishaev** [comment]
Could you please take a look at the PR? https://github.com/igrishaev/pg2/pull/16

**Comment by igrishaev** [comment]
Done, available in `[0.1.17]`
